aviatrix/resource_aviatrix_sumologic_forwarder.go

```diff
@@ -0,0 +1,124 @@
+package aviatrix
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aviatrix/goaviatrix"
+)
+
+func resourceAviatrixSumologicForwarder() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAviatrixSumologicForwarderCreate,
+		Read:   resourceAviatrixSumologicForwarderRead,
+		Update: resourceAviatrixSumologicForwarderUpdate,
+		Delete: resourceAviatrixSumologicForwarderDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"access_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Access ID",
+			},
+			"access_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Access key",
+			},
+			"source_category": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Source category",
+			},
+			"custom_cfg": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Custom cfg",
+			},
+			"excluded_gateways": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "List of excluded gateways.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Enabled or not.",
+			},
+		},
+	}
+}
+
+func marshalSumologicForwarderInput(d *schema.ResourceData) *goaviatrix.SumologicForwarder {
+	return &goaviatrix.SumologicForwarder{
+		AccessID:       d.Get("access_id").(string),
+		AccessKey:      d.Get("access_key").(string),
+		SourceCategory: d.Get("source_category").(string),
+		CustomCfg:      d.Get("custom_cfg").(string),
+	}
+}
+
+func resourceAviatrixSumologicForwarderCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	sumologicForwarder := marshalSumologicForwarderInput(d)
+
+	var excludedGateways []string
+	for _, v := range d.Get("excluded_gateways").(*schema.Set).List() {
+		excludedGateways = append(excludedGateways, v.(string))
+	}
+	if len(excludedGateways) != 0 {
+		sumologicForwarder.ExcludedGatewaysInput = strings.Join(excludedGateways, ",")
+	}
+
```

Comment:
This seems like it should be in `marshalSumologicForwarderInput` since it is also marshaling the ResourceData into your struct type right?